tests/common_dtd_z.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2013-2023 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ *
+ * @precisions normal z -> s d c
+ *
+ */
+#ifndef __COMMON_DTD_Z_H__
+#define __COMMON_DTD_Z_H__
+
+#include "common.h"
+#include "flops.h"
+#include "dplasma/types.h"
+#include "parsec/data_dist/matrix/sym_two_dim_rectangle_cyclic.h"
+#include "parsec/data_dist/matrix/two_dim_rectangle_cyclic.h"
+#include "parsec/interfaces/dtd/insert_function.h"
+
+
+#if defined(DPLASMA_HAVE_CUDA)
+#include "parsec/execution_stream.h"
+#include "parsec/parsec_internal.h"
+#include "parsec/mca/device/cuda/device_cuda.h"
+#include "parsec/utils/zone_malloc.h"
+#include "dplasmaaux.h"
+#include "potrf_cublas_utils.h"
+#include <cublas_v2.h>
+
+#endif
+
+int parsec_core_zpotrf(parsec_execution_stream_t *es, parsec_task_t *this_task);
+int parsec_core_zherk(parsec_execution_stream_t *es, parsec_task_t *this_task);
+int parsec_core_ztrsm(parsec_execution_stream_t *es, parsec_task_t *this_task);
+int parsec_core_zgemm(parsec_execution_stream_t *es, parsec_task_t *this_task);
+
+#if defined(DPLASMA_HAVE_CUDA)
+
+typedef struct zpotrf_dtd_workspace_info_s {
+    int mb;
+    int nb;
+    dplasma_enum_t uplo;
+} zpotrf_dtd_workspace_info_t;
+
+void* zpotrf_dtd_create_workspace(void *obj, void *user);
+void zpotrf_dtd_destroy_workspace(void *_ws, void *_n);
+
+int parsec_core_zpotrf_cuda(parsec_device_gpu_module_t* gpu_device,
+                            parsec_gpu_task_t*          gpu_task,
+                            parsec_gpu_exec_stream_t*   gpu_stream);
+int parsec_core_zherk_cuda( parsec_device_gpu_module_t* gpu_device,
+                            parsec_gpu_task_t*          gpu_task,
+                            parsec_gpu_exec_stream_t*   gpu_stream);
+int parsec_core_ztrsm_cuda( parsec_device_gpu_module_t* gpu_device,
+                            parsec_gpu_task_t*          gpu_task,
+                            parsec_gpu_exec_stream_t*   gpu_stream);
+int parsec_core_zgemm_cuda( parsec_device_gpu_module_t* gpu_device,
+                            parsec_gpu_task_t*          gpu_task,
+                            parsec_gpu_exec_stream_t*   gpu_stream);
+
+#endif /* DPLASMA_HAVE_CUDA */
+
+#define PASTE_CODE_CREATE_ZPOTRF_TASKCLASS(TASKPOOL, TASKCLASS)                                                        \
```

Comment:
I don't think these are doing any good, they make the code look simpler but less understandable. Readers need to see what they are doing, not have obscure macro sprinkled across the code, especially if there are very few uses.